### PR TITLE
Improved coin logic

### PR DESCRIPTION
**Pull request by MahdiBM**
This will improve the logic by which we allow users give coins to each other.
Now users can give coins to multiple users (like the old penny) and in more patterns.
Examples:
`xxxx @mahdi ++ xxxx` -> mahdi will get a new coin.
`xxxx @mahdi ++ xxxx @tim thanks! xxx` -> mahdi and tim will get a new coin.
`@mahdi xxxx :coin:` -> mahdi will get a new coin.
`xxxx @mahdi @benny thanks! xxxx` -> mahdi and benny will get a new coin.

### CODE REVIEW

**Comment by MahdiBM** [comment]
This PR now adds a lot more stuff than it was initially created for.

You will be able to give coins:
* Much more freely, with prefixing _or_ suffixing a coin-sign.
* Like before, with mentioning users at the beginning of a message and using a coin sign at the end of the message.
* To as many users as you want, in a single message. (currently limited to 10 to limit possible abuse)
* Only to real users.
* Not to channels or roles.
* To author of a replied message.
* And more...

The PR also comes with a lot of tests to make sure everything works as expected.

File: CODE/Sources/PennyBOT/Listeners/CoinHandler.swift [hunk 1]
```diff
@@ -0,0 +1,200 @@
+//
+//  File.swift
+//  
+//
+//  Created by Benny De Bock on 08/03/2022.
+//
+
```
Comment (0xTim) [inline CODE/Sources/PennyBOT/Listeners/CoinHandler.swift:7]: Lets remove this

File: CODE/Tests/PennyBOTTests/CoinHandlerTests.swift [hunk 2]
```diff
@@ -0,0 +1,477 @@
+//
+//  File.swift
+//  
+//
+//  Created by Mahdi Bahrami on 9/13/22.
+//
+
```
Comment (0xTim) [inline CODE/Tests/PennyBOTTests/CoinHandlerTests.swift:7]: Again remove

**Review by 0xTim — APPROVED** [review]
Looks great! A couple of file placeholders to remove and a conflict to fix then feel free to merge

**Pull request merged**